### ISSUE TITLE
Unnecessary performance bottleneck in Statevector.evolve

### ISSUE THREAD

**Issue by rumschuettel** [Qiskit/qiskit-terra#4281]
### Informations

- **Qiskit Aer version**: 0.5.0
- **Python version**: 3.8
- **Operating system**: linux (ubuntu 18.04)

### What is the current behavior?

Statevector.evolve is slow, because it recreates Statevector objects, which itself is a slow operation.

### Steps to reproduce the problem

Throw a line profiler on a code segment that calls Statevector.evolve repeatedly.

### What is the expected behavior?

Creating a Statevector should have little to no overhead.

### Suggested solutions

I profiled some code that uses Statevector.evolve repeatedly, and identified the bottleneck:

```
# qiskit/quantum_info/states/statevector.py in Statevector.evolve
return Statevector(np.reshape(tensor, np.product(new_dims)), dims=new_dims)
# same file, Statevector.__init__
super().__init__(...)
# qiskit/quantum_info/states/quantum_state.py in QuantumState.__init__
self._rng = np.random.RandomState()
```
Now, creating a RandomState object seems to have tremendous overhead; indeed, the np.reshape operation of tensor is much quicker.

Workaround: after loading qiskit and numpy, run
```
def _global_rng_fix():
    return _global_rng_fix._random_state
_global_rng_fix._random_state = np.random.RandomState()
np.random.RandomState = _global_rng_fix
```

This is not a fix! A true fix would be to use a newer rng object that doesn't have massive overhead (note that RandomState is legacy code anyhow - and since there's no seed set, it's unclear what the benefit of a separate RandomState object should be within a Statevector anyhow).

Thanks for your great work!
~ Johannes

**Comment by chriseclectic** [comment]
Nice find. Yes it seems like initializing the rng seed using `numpy.random.RandomState` in the base `QuantumState` class is the culprit and has huge overhead.

Changing it to just use `np.random` unless you user changes it with a fixed seed later I got timings:

```
# using self._rng = np.random.RandomState()
%timeit Statevector([1, 0])
162 µs ± 2.19 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

# using self._rng = np.random
%timeit Statevector([1, 0])
13.1 µs ± 155 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

I'll add this fix to #4223

**Comment by ajavadia** [comment]
is this fixed by https://github.com/Qiskit/qiskit-terra/pull/4297?
